### PR TITLE
Auth with fastapi users and GitHub, Orcid, Orcid sandbox

### PR DESCRIPTION
**Pull request by sverhoeven**
Implemented using [fastapi_users](https://fastapi-users.github.io/fastapi-users/) package.

Adds following routes (`/api/users/profile` route is written in this repo, rest of routes come from fastapi_users package):
localhost_8000_api_docs (1)

Other:
1. Run command renamed from `python -m bartender` to `bartender serve`.
2. Added `bartender super` sub command to make a user super user.
3. Renamed `id` file in job dir with `meta` and appended api token to it.
4. Require authentication for `PUT /api/job` route.

### CODE REVIEW

**Comment by sverhoeven** [comment]
There is also https://pypi.org/project/authx/ for user management, but https://authx.yezz.me/faq/#what-is-the-difference-between-the-fastapi-user-authenticationx says authx is more complex to use so decided to use fastapi_users package.

**Review by rvhonorato — APPROVED** [review]
Great!

I've tested it with ORCID and it works fine, with GitHub I'm getting a 404 when clicking the authorization url, after this;
```bash
curl -X 'GET' \
  'http://localhost:8000/auth/<name of social login>/authorize' \
  -H 'accept: application/json'
```

I've followed all the steps, so could you double check the instructions? 

No further comments on the code, all looks good to me.

File: bartender/__main__.py [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
 import uvicorn
 
+from bartender.db.dao.user_dao import get_user_db
+from bartender.db.session import make_engine, make_session_factory
 from bartender.settings import settings
 
 
-def main() -> None:
-    """Entrypoint of the application."""
+def serve() -> None:
```
Comment (rvhonorato) [inline bartender/__main__.py:14]: nice touch! 😄

**Comment by sverhoeven** [comment]
> Great!
> 
> I've tested it with ORCID and it works fine, with GitHub I'm getting a 404 when clicking the authorization url, after this;
> 
> ```shell
> curl -X 'GET' \
>   'http://localhost:8000/auth/<name of social login>/authorize' \
>   -H 'accept: application/json'
> ```
> 
> I've followed all the steps, so could you double check the instructions?
> 
> No further comments on the code, all looks good to me.

Thanks for testing and good that ORCID works.

Hmm GitHub works for me. 

Do the callback URL in GitHub app settings and the authorization URL have the same scheme, domain, port?
Do you have in `./.env` file the following?
```env
BARTENDER_GITHUB_CLIENT_ID=I...1e
BARTENDER_GITHUB_CLIENT_SECRET=fe...5
```

I got my authorization URL from `curl -X 'GET'  'http://localhost:8000/auth/github/authorize'  -H 'accept: application/json'` and it looks like
`https:\/\/github.com\/login\/oauth\/authorize?response_type=code&client_id=Iv1.fb2586d366ee1f1e&redirect_uri=http%3A%2F%2Flocalhost%3A8000%2Fauth%2Fgithub%2Fcallback&state=ey...1PgIM&scope=user+user%3Aemail` . 
After I load that in browser the response is like `{"access_token":"eyJ0...V8","token_type":"bearer"}`. The token can be pasted on https://jwt.io/ to decode it.

**Pull request merged**